packages/webpack-cli/lib/utils/arg-parser.js

```diff
@@ -90,6 +90,26 @@ const argParser = (options, args, argsOnly = false, name = '') => {
                 // a multiple argument parsing function
                 const multiArg = (value, previous = []) => previous.concat([value]);
                 parserInstance.option(flagsWithType, option.description, multiArg, option.defaultValue).action(() => {});
+            } else if (option.multipleType) {
+                // for options which accept multiple types like env
+                // so you can do `--env platform=staging --env production`
+                // { platform: "staging", production: true }
+                const multiArg = (value, previous = {}) => {
+                    const [allKeys, val] = value.split('=');
+                    const splitKeys = allKeys.split('.');
+                    let prevRef = previous;
+                    splitKeys.forEach((someKey, index) => {
+                        if (!prevRef[someKey]) prevRef[someKey] = {};
+                        if (index === splitKeys.length - 1) {
+                            prevRef[someKey] = val || true;
+                        } else {
+                            prevRef[someKey] = typeof prevRef[someKey] === 'string' ? {} : { ...prevRef[someKey] };
```

Comment:
This line seems hard to read because I don't think the nuances/side-effects of the spread operator are obvious (like what happens when `prevRef[someKey]` is neither object nor string -- it coerces to an empty object).

This might be easier to understand because it communicates the real intent to create a tree:

```javascript
// ensures that prefRef[someKey] is an object
if (null === prevRef[someKey] || 'object' !== typeof prevRef[someKey]) {
  prevRef[someKey] = {};
}
```

You might even be able to just get away with this since there are probably limits as to what `previous` can be (I'm just guessing, but don't really know):

```javascript
if ("string" === typeof prevRef[someKey]) {
  prevRef[someKey] = {};
}
```

Reply:
Ah right makes sense, thanks!